app/src/main/kotlin/com/getstrm/pace/processing_platforms/bigquery/BigQueryViewGenerator.kt

```diff
@@ -54,30 +54,34 @@ class BigQueryViewGenerator(
         } else {
             DSL.or(
                 principals.map { principal ->
-                    when {
-                        principal.hasGroup() ->
-                            if (useIamCheckExtension)
-                                DSL.condition(
-                                    "\"True\" in (select principal_check_routines.check_principal_access({0}))",
-                                    DSL.quotedName(principal.group)
-                                )
-                            else
-                                DSL.condition(
-                                    "{0} IN ( SELECT {1} FROM {2} )",
-                                    principal.group,
-                                    DSL.field(renderName("userGroup")),
-                                    DSL.field(renderName("user_groups"))
-                                )
-                        else ->
-                            throw InternalException(
-                                InternalException.Code.INTERNAL,
-                                DebugInfo.newBuilder()
-                                    .setDetail(
-                                        "Principal of type ${principal.principalCase} is not supported for platform BigQuery. ${PaceStatusException.UNIMPLEMENTED}"
+                    if (useIamCheckExtension) {
+                        val principalName =
+                            when {
+                                principal.hasGroup() -> DSL.quotedName("group:${principal.group}")
+                                principal.hasRole() -> DSL.quotedName("role:${principal.role}")
+                                principal.hasPermission() ->
+                                    DSL.quotedName("permission:${principal.permission}")
+                                else ->
+                                    throw InternalException(
+                                        InternalException.Code.INTERNAL,
+                                        DebugInfo.newBuilder()
+                                            .setDetail(
+                                                "Principal of type ${principal.principalCase.name} is not supported for platform BigQuery. ${PaceStatusException.UNIMPLEMENTED}"
+                                            )
+                                            .build()
                                     )
-                                    .build()
-                            )
-                    }
+                            }
+                        DSL.condition(
+                            "\"True\" in (select principal_check_routines.check_principal_access({0}))",
+                            principalName
+                        )
+                    } else
+                        DSL.condition(
+                            "{0} IN ( SELECT {1} FROM {2} )",
+                            principal.group,
```

Comment:
In the scenario where `useIamCheckExtension` is false, we must assert that `group` is used. Otherwise you can fill out one of the other `oneof` fields, which leads to incorrect SQL syntax: ` IN ( SELECT userGroup from user_groups)`,